### PR TITLE
feat (api): workflow conditions with lua script

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: sdk/luascript/luascript.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	} else {
+		lv := c.state.Get(-1)  // get the value at the top of the stack
+		if !lua.LVAsBool(lv) { // lv is neither nil nor false
+			ok = true
+		}
+	}
+
+	c.IsError = (err != nil)
+	c.Result = &ok
+	return
```
Comment (bnjjj) [inline sdk/luascript/luascript.go]: I think named return is less readable

File: sdk/luascript/luascript.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+// Check is a type which helps to call a lua script with variables to check something.
+// The lua script must return true/false
+type Check struct {
+	state                    *lua.LState
+	exceptionHandlerFunction *lua.LFunction
+	variables                map[string]string
+	IsError                  bool
+	Result                   *bool
```
Comment (bnjjj) [inline sdk/luascript/luascript.go]: Why use pointer here ?
cf: https://segment.com/blog/allocation-efficiency-in-high-performance-go-services/

File: sdk/workflow.go [hunk 3]
```diff
@@ -398,12 +398,18 @@ func (n *WorkflowNode) InvolvedEnvironments() []int64 {
...
-	WorkflowNodeID     int64                      `json:"workflow_node_id" db:"workflow_node_id"`
-	WorkflowDestNodeID int64                      `json:"workflow_dest_node_id" db:"workflow_dest_node_id"`
-	WorkflowDestNode   WorkflowNode               `json:"workflow_dest_node" db:"-"`
-	Conditions         []WorkflowTriggerCondition `json:"conditions,omitempty" db:"-"`
-	Manual             bool                       `json:"manual" db:"manual"`
+	ID                 int64                     `json:"id" db:"id"`
+	WorkflowNodeID     int64                     `json:"workflow_node_id" db:"workflow_node_id"`
+	WorkflowDestNodeID int64                     `json:"workflow_dest_node_id" db:"workflow_dest_node_id"`
+	WorkflowDestNode   WorkflowNode              `json:"workflow_dest_node" db:"-"`
+	Conditions         WorkflowTriggerConditions `json:"conditions,omitempty" db:"-"`
```
Comment (sguiheux) [inline sdk/workflow.go]: Missing UI impact

File: sdk/workflow.go [hunk 4]
```diff
@@ -398,12 +398,18 @@ func (n *WorkflowNode) InvolvedEnvironments() []int64 {
...
+	Manual             bool                      `json:"manual" db:"manual"`
+}
+
+//WorkflowTriggerConditions is either an array of WorkflowTriggerCondition or a lua script
+type WorkflowTriggerConditions struct {
```
Comment (sguiheux) [inline sdk/workflow.go]: Missing UI impact

File: sdk/workflow.go [hunk 5]
```diff
@@ -427,13 +433,13 @@ type WorkflowNodeContext struct {
...
-	WorkflowHookModelID int64                      `json:"workflow_hook_model_id" db:"workflow_hook_model_id"`
-	WorkflowHookModel   WorkflowHookModel          `json:"model" db:"-"`
-	Conditions          []WorkflowTriggerCondition `json:"conditions,omitempty" db:"-"`
-	Config              WorkflowNodeHookConfig     `json:"config" db:"-"`
+	ID                  int64                     `json:"id" db:"id"`
+	UUID                string                    `json:"uuid" db:"uuid"`
+	WorkflowNodeID      int64                     `json:"workflow_node_id" db:"workflow_node_id"`
+	WorkflowHookModelID int64                     `json:"workflow_hook_model_id" db:"workflow_hook_model_id"`
+	WorkflowHookModel   WorkflowHookModel         `json:"model" db:"-"`
+	Conditions          WorkflowTriggerConditions `json:"conditions,omitempty" db:"-"`
```
Comment (sguiheux) [inline sdk/workflow.go]: Missing UI impact

File: sdk/workflow.go (hunk 3 above)
Comment (fsamin) [reply sdk/workflow.go]: done

File: sdk/workflow.go (hunk 5 above)
Comment (fsamin) [reply sdk/workflow.go]: done

**Pull request merged**